src/ui/components/SecondaryToolbox/react-devtools/__tests__/rdtProcessing.test.ts

```diff
@@ -0,0 +1,332 @@
+  2,
+  // UTF-8 char codes: "B2"
+  66,
+  50,
+  // item 3 length
+  2,
+  // UTF-8 char codes: "C3"
+  67,
+  51,
+  // item 4 length
+  2,
+  // UTF-8 char codes: "D4"
+  68,
+  52,
+  // item 5 length
+  2,
+  // UTF-8 char codes: "E5"
+  69,
+  53,
+  // item 6 length
+  2,
+  // UTF-8 char codes: "k1",
+  107,
+  49,
+  // item 7 length
+  2,
+  // UTF-8 char codes: "k2"
+  107,
+  50,
+  // Operation 1: Add root, id 2
+  TREE_OPERATION_ADD,
+  2,
+  // Type of this node (root, function, class, memo, etc)
+  ElementTypeRoot,
+  // Strict mode compliant: false
+  0,
+  // supports profiling: false
+  0,
+  // supports strict mode: true
+  1,
+  // has owner metadata: true
+  1,
+  // Operation 2: Add standard node A1, id 3
+  TREE_OPERATION_ADD,
+  3,
+  ElementTypeFunction,
+  // parentId: root (2)
+  2,
+  // ownerID: root (2)
+  2,
+  // string table index for "A1" (index 0 is `null`)
+  1,
+  // key string table index (index 0 is `null`)
+  0,
+  // Operation 3: Add standard node B2, id 4
+  TREE_OPERATION_ADD,
+  4,
+  ElementTypeFunction,
+  // parentId: A1 (3)
+  3,
+  // ownerID: root (2)
+  2,
+  // string table index for "B2"
+  2,
+  // key string table index
+  0,
+  // Operation 4: Add standard node C3, id 5
+  TREE_OPERATION_ADD,
+  5,
+  ElementTypeFunction,
+  // parentId: root (2)
+  2,
+  // ownerID: root (2)
+  2,
+  // string table index for "C3"
+  3,
+  // key string table index
+  0,
+  // Operation 5: Add standard node D4, id 6
+  TREE_OPERATION_ADD,
+  6,
+  ElementTypeFunction,
+  // parentId: C3 (5)
+  5,
+  // ownerID: C3 (5)
+  5,
+  // string table index for "D4"
+  4,
+  // key string table index for "k1"
+  6,
+  // Operation 6: Add standard node E5, id 7
+  TREE_OPERATION_ADD,
+  7,
+  ElementTypeFunction,
+  // parentId: C3 (5)
+  5,
+  // ownerID: C3 (5)
+  5,
+  // string table index for "E5"
+  5,
+  // key string table index for "k2"
+  7,
+  // Operation 7: add errors/warnings to D4 (6)
+  TREE_OPERATION_UPDATE_ERRORS_OR_WARNINGS,
+  6,
+  // error count: 1
+  1,
+  // warning count: 2
+  2,
+  // Operation 8: reorder E5, D4, children of C3 (5)
+  TREE_OPERATION_REORDER_CHILDREN,
+  5,
+  // number of children: 2
+  2,
+  // child IDs: E5 (7), D4 (6)
+  7,
+  6,
+  // Operation 9: set subtree mode
+  TREE_OPERATION_SET_SUBTREE_MODE,
+  // rootId
+  2,
+  // mode: StrictMode (not that it matters here)
+  1,
+  // Operation 10: update tree base duration
+  // this is almost never going to get used, but let's be complete
+  TREE_OPERATION_UPDATE_TREE_BASE_DURATION,
+  // nodeId: C3 (5)
+  5,
+  // tree base duration: 1234
+  1234,
+  // Operation 11: remove B2 (4) and A1 (3)
+  TREE_OPERATION_REMOVE,
+  // number of nodes to remove
+  2,
+  // node IDs to remove: B2 (4), A1 (3), normally bottom-up
+  4,
+  3,
+  // Operation 12: remove root (2)
+  TREE_OPERATION_REMOVE_ROOT,
+  // No root ID specified - that was implicit because we specified
+  // the root ID for these ops as the second array index
+];
```

Comment:
I spiked this out and pushed as PR #9579

I think it caught a small bug in the serialization logic:
<img width="385" alt="Screen Shot 2023-08-03 at 10 16 32 AM" src="https://github.com/replayio/devtools/assets/29597/b5b2df94-34fa-4b9e-8996-09985348b02d">

Reply:
What is that change pointing to, operations-wise?

Reply:
Okay, I think I've figured out it's actually `supportsProfiling` for adding a root. I assumed it was also a boolean, but per the RDT source it can carry more info:

```ts
    // Adding a new field here would require a bridge protocol version bump (a backwads breaking change).
    // Instead let's re-purpose a pre-existing field to carry more information.
    let profilingFlags = 0;
    if (isProfilingSupported) {
      profilingFlags = PROFILING_FLAG_BASIC_SUPPORT;
      if (typeof injectProfilingHooks === 'function') {
        profilingFlags |= PROFILING_FLAG_TIMELINE_SUPPORT;
      }
    }

    if (isRoot) {
      pushOperation(TREE_OPERATION_ADD);
      pushOperation(id);
      pushOperation(ElementTypeRoot);
      pushOperation((fiber.mode & StrictModeBits) !== 0 ? 1 : 0);
      pushOperation(profilingFlags);
      pushOperation(StrictModeBits !== 0 ? 1 : 0);
      pushOperation(hasOwnerMetadata ? 1 : 0);
```